bindings/go/runtime/registry.go

```diff
@@ -154,7 +128,7 @@ func (r *Scheme) NewObject(typ Type) (any, error) {
 		}
 		object = reflect.New(t).Interface()
 
-		return object, nil
+		return object.(Typed), nil
```

Comment:
Type check here?

Reply:
if the object is created through reflection above and we did an assertion through the the registry during creation, how would this type check be needed?

Reply:
Ah you're right, sorry the diff was blurring out that section of the code and I didn't check.